src/Data/Managers/V2/PromoteVersion/usePromoteVersion.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  UseMutationResult,
+  useMutation,
+  useQueryClient,
+} from "@tanstack/react-query";
+import { PrimaryBaseUrlManager } from "@/UI";
+
+import { useFetchHelpers } from "../helpers";
+
+/**
+ * React Query hook for promoting a version of desired state
+ *
+ * @param {string} env - The environment in which we are trying to promote the version.
+ * @returns {Mutation} The mutation object for sending the request.
+ */
+export const usePromoteVersion = (
+  environment: string,
+): UseMutationResult<void, Error, string, unknown> => {
+  const client = useQueryClient();
+  const { createHeaders, handleErrors } = useFetchHelpers();
+  const headers = createHeaders(environment);
+  const baseUrlManager = new PrimaryBaseUrlManager(
+    globalThis.location.origin,
+    globalThis.location.pathname,
+  );
+  const baseUrl = baseUrlManager.getBaseUrl(process.env.API_BASEURL);
+
+  /**
+   * Sends a request to promote a version of desired state
+   * @param {string} version - the stringified version of desired state.
+   * @throws {Error} If the response is not successful, an error with the error message is thrown.
```

Comment:
Can you add this in the other places too? I notice some fetchers in the other queries don't have docstrings, or that they are not aligned.